packages/@lwc/synthetic-shadow/src/faux-shadow/portal.ts

```diff
@@ -40,17 +40,35 @@ function adoptChildNode(node: Node, fn: ShadowRootResolver, shadowToken: string
     }
 }
 
+function disownChildNode(node: Node) {
+    // Note: this will remove the node owner key.
+    setShadowRootResolver(node, function() {} as ShadowRootResolver);
+
+    if (node instanceof Element) {
+        setShadowToken(node, undefined);
+        const childNodes = getInternalChildNodes(node);
+        for (let i = 0, len = childNodes.length; i < len; i += 1) {
+            const child = childNodes[i];
+            disownChildNode(child);
+        }
+    }
+}
+
 function initPortalObserver() {
     return new MO(mutations => {
         forEach.call(mutations, mutation => {
-            const { target: elm, addedNodes } = mutation;
+            const { target: elm, addedNodes, removedNodes } = mutation;
             // the target of the mutation should always have a ShadowRootResolver attached to it
             const fn = getShadowRootResolver(elm) as ShadowRootResolver;
             const shadowToken = getShadowToken(elm);
             for (let i = 0, len = addedNodes.length; i < len; i += 1) {
                 const node: Node = addedNodes[i];
                 adoptChildNode(node, fn, shadowToken);
             }
+
+            for (let i = 0, len = removedNodes.length; i < len; i += 1) {
```

Comment:
you also have the be very careful here because removed and reinserted elements will have issues... I think you will have to move this block before the adaptChildNode for-loop, just in case.